### PR TITLE
Remember deployed tags

### PR DESCRIPTION
**Pull request by dgageot**

### CODE REVIEW

File: pkg/skaffold/deploy/helm.go [hunk 1]
```diff
@@ -29,16 +29,14 @@ import (
 )
 
 type HelmDeployer struct {
-	*config.DeployConfig
-	kubeContext string
+	*KubectlDeployer
```
Comment (r2d4) [inline pkg/skaffold/deploy/helm.go]: Why does HelmDeployer embed KubectlDeployer?

File: pkg/skaffold/deploy/kubectl.go [hunk 2]
```diff
@@ -65,6 +65,8 @@ spec:
 
 type KubectlDeployer struct {
 	*config.DeployConfig
+
+	buildTags   *BuildTags
```
Comment (r2d4) [inline pkg/skaffold/deploy/kubectl.go]: Should buildTags go on SkaffoldRunner instead?

File: pkg/skaffold/deploy/helm.go (hunk 1 above)
Comment (dgageot) [reply pkg/skaffold/deploy/helm.go]: I thought it made sense but I'll revert it

File: pkg/skaffold/deploy/kubectl.go (hunk 2 above)
Comment (dgageot) [reply pkg/skaffold/deploy/kubectl.go]: I can try but I'm not sure it makes sense

**Comment by dgageot** [comment]
@r2d4 Should be better now. I realized I'm not a big fan of passing a `BuildResult` to a `Deployer`.

**Comment by r2d4** [comment]
Fixes #193

File: pkg/skaffold/runner/runner.go [hunk 3]
```diff
@@ -166,24 +167,69 @@ func (r *SkaffoldRunner) dev(artifacts []*config.Artifact) error {
...
 	}
+
 	fmt.Fprintln(r.opts.Output, "Deploy complete in", time.Now().Sub(start))
 
-	return bRes, nil, nil
+	return dRes, nil
+}
+
+func mergeWithPreviousBuilds(builds, previous []build.Build) []build.Build {
+	updatedBuilds := map[string]bool{}
```
Comment (r2d4) [inline pkg/skaffold/runner/runner.go]: any issue with just putting map[string]build.Build on SkaffoldRunner instead of the slice?

Comment (dgageot) [reply pkg/skaffold/runner/runner.go]: no, no issue. Do you think it would be easier/clearer?

**Comment by dgageot** [comment]
@r2d4 I've rebased the PR

File: pkg/skaffold/runner/runner.go (hunk 3 above)
Comment (r2d4) [reply pkg/skaffold/runner/runner.go]: Nope, might be slightly cleaner, but shouldn't really make much of a difference.

**Pull request merged**